### PR TITLE
Added methods to get value expression

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/ReflectionEnumCase.php [hunk 1]
```diff
@@ -86,6 +86,16 @@ public function getName(): string
         return $this->name;
     }
 
+    /** @throws LogicException */
+    public function getValueExpression(): Node\Expr
+    {
+        if ($this->value === null) {
+            throw new LogicException('This enum case does not have a value');
+        }
```
Comment (Ocramius) [inline src/Reflection/ReflectionEnumCase.php]: Hmm, this API is a bit "meh".

How does a consumer prevent calling this upfront, if no value exists?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Reflection/ReflectionEnumCase.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionEnumCase.php]: I did the same as https://github.com/Roave/BetterReflection/blob/6.0.x/src/Reflection/ReflectionEnumCase.php#L100

Consumer should probably check `ReflectionEnum::isBacked()`

Comment (Ocramius) [reply src/Reflection/ReflectionEnumCase.php]: Let's add that to the docblock as documentation 👍

Comment (kukulich) [reply src/Reflection/ReflectionEnumCase.php]: Added.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Pull request merged**